app/audit/models/ClaimARefundAuditModel.scala

```diff
@@ -43,19 +44,15 @@ case class ClaimARefundAuditModel(balanceDetails: Option[BalanceDetails],
   }
 
   private def getCreditType(credit: (DocumentDetailWithDueDate, FinancialDetail)): String = {
-    val isMFA: Boolean = credit._2.isMFACredit
-    val isCutOverCredit: Boolean = credit._2.isCutOverCredit
+    val creditType: Option[CreditType] = credit._2.getCreditType
     val isPayment: Boolean = credit._1.documentDetail.paymentLot.isDefined
-    val isBalancingCreditCharge: Boolean = credit._2.isBalancingChargeCredit
-
-    (isMFA, isCutOverCredit, isBalancingCreditCharge, isPayment) match {
-      case (true, _, _, _) => "Credit from HMRC adjustment"
-      case (_, true, _, _) => "Credit from an earlier tax year"
-      case (_, _, true, _) => "Balancing charge credit"
-      case (_, _, _, true) => s"Payment made on ${getFullDueDate(credit._1.dueDate.get)}"
-
-      case error =>
-        Logger("application").error(s"[ClaimARefundAuditModel][getCreditType] Missing or non-matching credit: $error found")
+    creditType match {
+      case Some(MfaCreditType) => "Credit from HMRC adjustment"
+      case Some(CutOverCreditType) => "Credit from an earlier tax year"
+      case Some(BalancingChargeCreditType) => "Balancing charge credit"
+      case _ if isPayment => s"Payment made on ${getFullDueDate(credit._1.dueDate.get)}"
```

Comment:
its not part of you changes, but we should really avoid calling .get directly, might be better to call something like .getOrElse(LocaDate.Min), so you would return 'beginning of time' instead , and if data is wrong we not fail this part;